tools/mason/MasonExternal.chpl

```diff
@@ -261,10 +261,10 @@ proc getSpackVersion : VersionInfo {
   const command = "spack --version";
   const tmpVersion = getSpackResult(command,true).strip();
   // on systems with their own spack, spack --version can provide
-  // a version string like x.x.x-xxxx-hash
+  // a version string like x.x.x.xxxx (hash)
   // partitioning the string allows us to separate the major.minor.bug
   // from the remaining values
-  const version = tmpVersion.partition("-");
+  const version = tmpVersion.partition(" ");
```

Comment:
Elliot I see this is the same change as the one I did in https://github.com/chapel-lang/chapel/pull/21439/files but I think my version allows both formats. Do you think that we should allow both version formats here or let the old format go away & say that `mason external` only supports new enough spack versions?

Reply:
In your PR, I think `nodash` was unused, but in general I'm good saying we only support new enough spack versions.